Modules/_hashopenssl.c

```diff
@@ -1682,24 +1681,24 @@ pbkdf2_hmac_impl(PyObject *module, const char *hash_name,
         goto end;
     }
 
-    key_obj = PyBytes_FromStringAndSize(NULL, dklen);
-    if (key_obj == NULL) {
-        goto end;
+    PyBytesWriter *writer = PyBytesWriter_Create(dklen);
+    if (writer == NULL) {
+        return NULL;
```

Comment:
```suggestion
        goto end;
```

`digest` is leaked otherwise.

Reply:
Oops, well spotted.